cookbooks/aws-parallelcluster-environment/test/controls/login_nodes_keys_spec.rb

```diff
@@ -0,0 +1,93 @@
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+keys_manager_script_dir = "/opt/parallelcluster/scripts/login_nodes"
+key_types = %w(ecdsa ed25519)
+is_not_amazon = !os_properties.amazon_family?
+is_ubuntu = os_properties.ubuntu?
+if is_not_amazon
+  key_types << 'rsa'
+end
+if is_ubuntu
+  key_types << 'dsa'
+end
+
+control 'head_node_directory_initialized' do
+  only_if { instance.head_node? && node['cluster']['scheduler'] != 'awsbatch' }
```

Comment:
[Nitpick] Like you did for other checks you can define `not_awsbatch = node['cluster']['scheduler'] != 'awsbatch' `

and use here and below `.. && not_awsbatch`

Reply:
This change requires an extra full test. Since it's not a blocking comment I'd prefer to skip this change.